html5ever/macros/match_token.rs

```diff
@@ -154,82 +155,89 @@ pub struct Tag {
     name: Option<syn::Ident>,
 }
 
-impl syn::synom::Synom for Tag {
-    named!(parse -> Self, do_parse!(
-        punct!(<) >>
-        closing: option!(punct!(/)) >>
-        name: alt!(
-            syn!(syn::Ident) => { |i| Some(i) }
-            |
-            punct!(_) => { |_| None }
-        ) >>
-        punct!(>) >>
-        (
-            Tag {
+impl Parse for Tag {
+    fn parse(input: ParseStream) -> Result<Self> {
+        input.parse::<Token![<]>()?;
+        let closing = input.parse::<Option<Token![/]>>()?;
```

Comment:
I would write this with the type on the left because `![/]>>()?;` is over the threshold of reasonable adjacent punctuation.

```rust
let closing: Option<Token![/]> = input.parse()?;
```